pkg/pillar/cmd/zedagent/handlelocation.go

```diff
@@ -59,14 +60,17 @@ func locationTimerTask(ctx *zedagentContext, handleChannel chan interface{}) {
 	for {
 		select {
 		case <-cloudTicker.C:
+			// Loopback the destination, will be fetched on the next iteration
+			triggerCloudLocationInfo <- AllDest
```

Comment:
How do you know that this can not block?
If the triggerPublishLocationToControllerWithDest() sent one message to the channel and then before that is processed this function handles a ticker, then it will block until the channel is read, but only this function reads the channel.

One way to solve this is to make this a conditional send but not clear to me that will always use the correct set of destinations.